include/NeoFOAM/core/parallelAlgorithms.hpp

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 NeoFOAM authors
+#pragma once
+
+#include <Kokkos_Core.hpp>
+#include "NeoFOAM/core/executor/executor.hpp"
+#include <type_traits>
+
+namespace NeoFOAM
+{
+
+template<typename executor, typename Kernel>
+void parallelFor(const executor& exec, label start, label end, Kernel kernel)
```

Comment:
i would prefer passing start and end together and until we have a NeoFOAM::Range just pass as `std::pair`

```suggestion
void parallelFor(const executor& exec, std::pair<size_t,size_t> range, Kernel kernel)
```

The reasoning is that start and ends are often defined by a patch which already has a .range() member.

Reply:
you can then unpack it `auto [start, end] = range`. If we implement a Range as struct we can use range.start and range.end.